packages/flutter/test/rendering/view_chrome_style_test.dart

```diff
@@ -0,0 +1,255 @@
+          await tester.pumpWidget(const SizedBox.expand());
+          await tester.pumpAndSettle();
+
+          expect(SystemChrome.latestStyle?.systemNavigationBarColor, isNull);
+        },
+        variant: TargetPlatformVariant.only(TargetPlatform.android),
+      );
+
+      testWidgets(
+        'systemNavigationBarColor is set for annotated view',
+        (WidgetTester tester) async {
+          setupTestDevice();
+          await tester.pumpWidget(const AnnotatedRegion<SystemUiOverlayStyle>(
+            value: SystemUiOverlayStyle(
+              systemNavigationBarColor: Colors.blue,
+            ),
+            child: SizedBox.expand(),
+          ));
+          await tester.pumpAndSettle();
+
+          expect(
+            SystemChrome.latestStyle?.systemNavigationBarColor,
+            Colors.blue,
+          );
+        },
+        variant: TargetPlatformVariant.only(TargetPlatform.android),
+      );
+
+      testWidgets(
+        'systemNavigationBarColor isn\'t set when view covers less than half of navigation bar',
+        (WidgetTester tester) async {
+          setupTestDevice();
+          const double lessThanHalfOfTheNavigationBarHeight =
+              _navigationBarHeight / 2.0 - 1;
+          await tester.pumpWidget(const Align(
+            alignment: Alignment.bottomCenter,
+            child: AnnotatedRegion<SystemUiOverlayStyle>(
+              value: SystemUiOverlayStyle(
+                systemNavigationBarColor: Colors.blue,
+              ),
+              child: SizedBox(
+                width: 100,
+                height: lessThanHalfOfTheNavigationBarHeight,
+              ),
+            ),
+          ));
+          await tester.pumpAndSettle();
+
+          expect(SystemChrome.latestStyle?.systemNavigationBarColor, isNull);
+        },
+        variant: TargetPlatformVariant.only(TargetPlatform.android),
+      );
+
+      testWidgets(
+        'systemNavigationBarColor is set when view covers more than half of navigation bar',
+        (WidgetTester tester) async {
+          setupTestDevice();
+          const double moreThanHalfOfTheNavigationBarHeight =
+              _navigationBarHeight / 2.0 + 1;
+          await tester.pumpWidget(const Align(
+            alignment: Alignment.bottomCenter,
+            child: AnnotatedRegion<SystemUiOverlayStyle>(
+              value: SystemUiOverlayStyle(
+                systemNavigationBarColor: Colors.blue,
+              ),
+              child: SizedBox(
+                width: 100,
+                height: moreThanHalfOfTheNavigationBarHeight,
+              ),
+            ),
+          ));
+          await tester.pumpAndSettle();
+
+          expect(
+            SystemChrome.latestStyle?.systemNavigationBarColor,
+            Colors.blue,
+          );
+        },
+        variant: TargetPlatformVariant.only(TargetPlatform.android),
+      );
+    });
+  });
+}
+
+class MockWindowPadding implements WindowPadding {
```

Comment:
nit: call this `FakeWindowPadding`, since `Mock_` is often used for classes which depend on `mockito`